### PR TITLE
fix: bug in api.ts when relation is null / missing meta

### PR DESCRIPTION
**Pull request by barakplasma**
feat: update dependencies for typescript, directus-extension, etc
feat: improve typescript types to detect quite a few problems (check `code.vue` L27; no "types")
fix: bug in `api.ts` when relation is null

BTW, I was unable to use `./dev.sh` but I copied the index.js into my directus instance to test the fix.

### CODE REVIEW

File: src/lib/api.ts [hunk 1]
```diff
@@ -38,22 +38,30 @@ export async function getCollections(api) {
   );
   const relations = relationsRes.data.data;
   relations.forEach((relation) => {
-    const oneField = collections[relation.meta.one_collection]?.fields.find(
```
Comment (barakplasma) [inline src/lib/api.ts:41]: my bug was `relation.meta.one_collection` being null, so this extension wouldn't work.

File: tsconfig.json [hunk 2]
```diff
@@ -1,11 +1,8 @@
 {
+  "extends": "@tsconfig/nuxt/tsconfig.json",
```
Comment (barakplasma) [inline tsconfig.json:2]: latest vue related types I could find

**Comment by barakplasma** [comment]
btw, see also https://github.com/directus-community/awesome-directus/pull/30 where I'm adding this extension to awesome-directus

File: package-lock.json
Comment (maltejur) [inline package-lock.json:1]: I don't think we should have two lockfiles at the same time, could you remove this one?

File: package.json [hunk 3]
```diff
@@ -11,7 +11,7 @@
     "type": "module",
     "path": "dist/index.js",
     "source": "src/index.ts",
-    "host": "^9.0.0-rc.95",
```
Comment (maltejur) [inline package.json:14]: This would mean the addon doesn't work anymore on old directus installation, which (as far as I know), it still does.

**Review by maltejur — CHANGES_REQUESTED** [review]
Thanks a lot for the MR, and also for the one on awesome-directus :D 

> BTW, I was unable to use ./dev.sh but I copied the index.js into my directus instance to test the fix.

Are you on Windows? There of course bash is not supported by default. If you are on Linux/macOS, what error are you getting?

File: package.json (hunk 3 above)
Comment (barakplasma) [reply package.json]: reverted it back to support v9.0, I just didn't test on it.

File: package.json [hunk 4]
```diff
@@ -20,15 +20,18 @@
...
-    "typescript": "^4.8.4",
-    "vue": "^3.2.40"
+    "axios": "^1.3.6",
+    "sass": "^1.62.0",
+    "typescript": "^5.0.4",
+    "vue": "^3.2.47"
   },
   "dependencies": {
-    "@directus/shared": "^9.18.1",
+    "@directus/shared": "^9.24.0",
```
Comment (barakplasma) [inline package.json:34]: I'm worried that if we use the directus shared types for a version after 9.0, then the directus host compatibility set at 9.0 might not actually be compatible.

**Comment by barakplasma** [comment]
I had trouble running ./dev.sh on Ubuntu 22.04; the script ran in bash, but didn't copy the built extension into the directus instance for me. I don't recall getting any error specifically, but we can chalk it up to environment differences. I could contribute a [devcontainer](https://code.visualstudio.com/docs/devcontainers/create-dev-container) if you like.

**Comment by barakplasma** [comment]
I'm closing this since I won't have time to continue working on it.

**Pull request closed without merging**